### DISCUSSION TITLE
How to run migrations in production. All commands are --preview-feature AKA not production ready?

### DISCUSSION THREAD

**Discussion by bobbyflowstate** [prisma/prisma#5166] [Q&A]
On this page: https://www.prisma.io/docs/concepts/components/prisma-migrate, it states:
_The prisma migrate dev --preview-feature and prisma migrate reset --preview-feature commands are designed to be used in development only, and should not affect production data._

On this page: , https://www.prisma.io/docs/concepts/components/prisma-migrate/prisma-migrate-flows it states:
_Use the following command in non-development environments to apply migrations:
`prisma migrate deploy --preview-feature`_

**What is the best practice for migrating production database?**
Firstly, I want to clarify that `prisma migrate deploy --preview-feature` is the command I want to use for production? I've tried this, however it's still pointing to my local database (DATABASE_URL in `.env` file) instead of my production database (DATABASE_URL in `production.env` file). I worked around this by running `DATABASE_URL=<prod_url> npx prisma migrate deploy --preview-feature`.  Is this the best practice?

**How is Prisma production ready?**
My concern is that the tool used to migrate production database is currently still in `preview-feature` mode. Yet Prisma website states Prisma is ready for production use. Am I missing something here? What is the best practice to migrate production databases? I have been using knex.js for some time and I feel confident in their tool to move `up` and `down` migrations. What is the equivalent here?

**Comment by ryands17** [comment]
Hey @rajuashok 👋 

Currently Prisma Migrate is in preview but we will be releasing a production ready release soon with [Native Types](https://github.com/prisma/prisma/issues/4330) support so currently you can use it in production as you can directly edit the `.sql` files generated by Migrate which is similar to Knex in this case.

> What is the best practice for migrating production database?

The best practice for performing migrations on your production database is to run the following command:

```
prisma migrate deploy --preview-feature
```

Currently Prisma only supports the `.env` file and not `production.env` so you would need to keep your URL in that file itself.

> How is Prisma production ready?

Prisma has 2 parts: [Prisma Client](https://www.prisma.io/docs/concepts/components/prisma-client) and Prisma Migrate. Both of these do not need any dependency on each other as Prisma Client is to query your database and Migrate is for Migrations. Prisma Migrate is going to be released soon for production, so if you are using Knex.js you can continue using it with Prisma and just use Prisma to query your database.

As for `up` and `down` migrations, we do not support `down` migrations at the moment as we are searching for the best use case and flow where they actually could be performed in a useful manner. You can read more about this [here](https://www.notion.so/prismaio/Documentation-for-Prisma-Migrate-Early-Access-7708df0c1d3d4852ab39fb69ed182786).

Let me know if that helps :)

Let me know if that

**Reply by bobbyflowstate** [reply]
Thank you so much for the in-depth response. This is helpful. Excited to hear prisma migrate will soon be production ready. Also helpful to know that I can use knex migrations in the meantime while also taking advantage of prisma client. Really great choice in keeping migrate and client separate.

Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
